clients/src/main/java/org/apache/kafka/clients/consumer/internals/Coordinator.java

```diff
@@ -49,38 +43,30 @@
 
 import java.io.Closeable;
 import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
 import java.util.HashMap;
 import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
-import java.util.concurrent.TimeUnit;
 
 /**
  * This class manages the coordination process with the consumer coordinator.
  */
-public final class Coordinator implements Closeable {
+public final class Coordinator extends GroupCoordinator<ConsumerProtocol.Subscription, ConsumerProtocol.Assignment>
```

Comment:
A general comment: maybe we can rename GroupCoordinator to Coordinator, and Coordinator to ConsumerCoordinator, and the server-side ConsumerCoordiantor to GroupCoordinator. Do you think the names are more clear that way?